datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -131,8 +192,29 @@ impl JoinLeftData {
     /// Decrements the counter of running threads, and returns `true`
     /// if caller is the last running thread
     fn report_probe_completed(&self) -> bool {
-        self.probe_threads_counter.fetch_sub(1, Ordering::Relaxed) == 1
+        self.probe_threads_counter.load(Ordering::Relaxed) == 0
+            || self.probe_threads_counter.fetch_sub(1, Ordering::Relaxed) == 1
+    }
+}
+
+fn merge_bitmap(m1: &mut BooleanBufferBuilder, m2: BooleanBuffer) -> Result<()> {
```

Comment:
bitmap here is boolean bitmask of what was visited or matched by join/filter?

Reply:
Correct. What will be shared and merged is the `JoinLeftData::visited_indices_bitmap`